PWGMM/Mult/Tasks/dndeta-hi.cxx

```diff
@@ -318,551 +332,567 @@ struct MultiplicityCounter {
+        }
+      }
+    }
   }
-  PROCESS_SWITCH(MultiplicityCounter, processV0CountingWithoutCent, "V0Count tracks without Centrality", false);
+  PROCESS_SWITCH(MultiplicityCounter, processV0Counting, "MC V0Count tracks without Centrality", false);
 
-  void processMCV0CountingWithoutCent(
-    soa::Join<MyCollisions, aod::McCollisionLabels> const& collisions,
+  void processMCV0Counting(
+    soa::Join<MyCollisionsCent, aod::McCollisionLabels> const& collisions,
     aod::McCollisions const& MCCollisions,
     Particles const& mcParticles,
     soa::Filtered<aod::V0Datas> const& fullV0s,
     soa::Filtered<LabeledTracksEx> const& tracks,
-    DaughterTracks const& Dautrks,
-    soa::SmallGroups<aod::ReassignedTracksCore> const& atracks)
+    DaughterTracks const& Dautrks)
   {
-    runMCV0Counting(collisions, MCCollisions, mcParticles, fullV0s, tracks, Dautrks, atracks);
-  }
-  PROCESS_SWITCH(MultiplicityCounter, processMCV0CountingWithoutCent, "MC V0Count tracks without Centrality", false);
+    for (auto& collision : collisions) {
+      auto cent = -1.0;
 
-  void processMCCountingWithoutCent(
-    soa::Join<MyCollisions, aod::McCollisionLabels> const& collisions,
-    aod::McCollisions const& MCCollisions,
-    Particles const& mcParticles,
-    soa::Filtered<LabeledTracksEx> const& Fitrks,
-    DaughterTracks const& Dautrks,
-    soa::SmallGroups<aod::ReassignedTracksCore> const& atracks,
-    soa::Join<aod::MFTTracks, aod::McMFTTrackLabels> const& mfttracks)
-  {
-    // runMCCounting(collisions, MCCollisions, fullV0s, mcParticles, Fitrks, Dautrks, atracks, mfttracks);
-    runMCCounting(collisions, MCCollisions, mcParticles, Fitrks, Dautrks, atracks, mfttracks);
-  }
-  PROCESS_SWITCH(MultiplicityCounter, processMCCountingWithoutCent, "MC Count tracks", false);
+      if (IsPbPb) {
+        if constexpr (MyCollisionsCent::template contains<aod::CentFT0Cs>())
+          cent = collision.centFT0C();
+      } else {
+        if constexpr (MyCollisionsCent::template contains<aod::CentFT0Ms>())
+          cent = collision.centFT0M();
+      }
 
-  void processGen(
-    aod::McCollisions::iterator const& mcCollision,
-    o2::soa::SmallGroups<soa::Join<aod::Collisions, aod::EvSels, aod::McCollisionLabels>> const& collisions,
-    Particles const& mcParticles, FiTracks const& tracks)
-  {
-    auto perCollisionMCSample = mcSample->sliceByCached(aod::mcparticle::mcCollisionId, mcCollision.globalIndex(), cache);
-    auto genz = mcCollision.posZ();
-    registry.fill(HIST("Tracks/ProcessGen/hgenzvtx"), Double_t(kINEL), genz);
-    for (auto& particle : perCollisionMCSample) {
-      auto p = pdg->GetParticle(particle.pdgCode());
-      if (p != nullptr) {
-        if (std::abs(p->Charge()) >= 3) {
-          registry.fill(HIST("Tracks/ProcessGen/hgendndeta"), Double_t(kINEL), genz, particle.eta());
+      auto mcCollision = collision.mcCollision();
```

Comment:
This should be after `has_mcCollision()` is checked. And this variable is not actually used in the code below, is it a mistake?

Reply:
I think it will be used later..